pinot-core/src/main/java/org/apache/pinot/core/operator/filter/AndFilterOperator.java

```diff
@@ -53,7 +53,12 @@ protected BlockDocIdSet getTrues() {
     Tracing.activeRecording().setNumChildren(_filterOperators.size());
     List<BlockDocIdSet> blockDocIdSets = new ArrayList<>(_filterOperators.size());
     for (BaseFilterOperator filterOperator : _filterOperators) {
-      blockDocIdSets.add(filterOperator.getTrues());
+      BlockDocIdSet blockDocIdSet = filterOperator.getTrues();
+      blockDocIdSets.add(blockDocIdSet);
+      if (blockDocIdSet.isAlwaysFalse()) {
+        // Return AndDocIdSet to ensure that getNumEntriesScannedInFilter is correctly reported.
+        return new AndDocIdSet(blockDocIdSets, _queryOptions, true);
```

Comment:
Do we need to create `AndDocIdSet` here? The scan shouldn't happen yet. The overhead of using `AndDocIdSet` is quite high